pkg/ui/react-app/src/pages/graph/Panel.tsx

```diff
@@ -255,8 +268,12 @@ class Panel extends Component<PanelProps & PathPrefixProps, PanelState> {
     this.setOptions({ usePartialResponse: event.target.checked });
   };
 
+  handleStoreMatchChange = (selectedStores: any): void => {
+    this.setOptions({ storeMatches: selectedStores });
```

Comment:
Oh, I found a non-obvious bug here. If you add some stores to the filter and then remove all of them `selectedStores` becomes `null` instead of `[]`.

Modify this line to `this.setOptions({ storeMatches: selectedStores || [] });`

Also, just to be on the safer side, change line 152 to expect `storeMatchers` as `null`. Can be done easily using optional chaining; `    this.props.options.storeMatches?.forEach((store: Store) =>` (notice the `?` after `storeMatches`).

Reply:
Fixed! That one line check using `?` was really cool! :)

Reply:
You forget to run `make assets` though :sweat_smile: so the production build is still buggy.